### ISSUE TITLE
"Author" User role has a useless shortcode generator on pages and posts

### ISSUE THREAD

**Issue by Benunc** [WordImpress/Give#845]
Current issue: if you create a User role of "Author," where the user has richediting enabled, Give is currently outputting a shortcode generator, but since they don't have the ability to see or edit give_forms, their shortcode generator is non-functional.

#842 addresses this

**Comment by mathetos** [comment]
I don't think the answer to this question is to prevent them from embedding the Give form (which by the looks of #849 is what it does). Instead, I think the generator should be made functional for them. 

Basically, there's no reason to limit who can embed Give forms from our end. The user can customize that however they want, but our roles/capabilities shouldn't be focused on adding forms to posts/pages.

I know this is merged already, but I'm confused on it's actual utility.

**Comment by Benunc** [comment]
The problem is that currently the actual generator for the shortcode
doesn't work, because it checks for greater privileges to run the
JavaScript.

On Aug 8, 2016 6:29 PM, "Matt Cromwell" <notifications@github.com> wrote:

I don't think the answer to this question is to prevent them from embedding
the Give form (which by the looks of #849
<https://github.com/WordImpress/Give/pull/849> is what it does). Instead, I
think the generator should be made functional for them.

Basically, there's no reason to limit who can embed Give forms from our
end. The user can customize that however they want, but our
roles/capabilities shouldn't be focused on adding forms to posts/pages.

I know this is merged already, but I'm confused on it's actual utility.

—
You are receiving this because you were assigned.

Reply to this email directly, view it on GitHub
<https://github.com/WordImpress/Give/issues/845#issuecomment-238397687>,
or mute
the thread

**Comment by mathetos** [comment]
Ya, then my suggestion is to fix the generator rather than exclude the author.

**Comment by DevinWalker** [comment]
Here's the permissions Woo uses: https://github.com/woothemes/woocommerce-shortcodes/blob/master/includes/class-wc-shortcodes-admin.php#L24-L33
